pipeline/trusted-resources/pkg/webhook/validation_admit_test.go

```diff
@@ -0,0 +1,364 @@
+		taskRun: taskRunContentTempered,
+		wantErr: true,
+	}, {
+		name:    "API Task Fail Verification without signature",
+		taskRun: taskRunUnsigned,
+		wantErr: true,
+	}, {
+		name:    "OCI Bundle Pass Verification",
+		taskRun: taskRunPassOCIBundle,
+		wantErr: false,
+	}, {
+		name:    "OCI Bundle Fail Verification without tempered conten",
+		taskRun: taskRunPassOCIBundleContentTempered,
+		wantErr: true,
+	}, {
+		name:    "OCI Bundle Fail Verification without signature",
+		taskRun: taskRunPassOCIBundleUnsigned,
+		wantErr: true,
+	}, {
+		name:    "Local taskRef Pass Verification",
+		taskRun: taskRunPassTaskRef,
+		wantErr: false,
+	}, {
+		name:    "Local taskRef Fail Verification with tempered content",
+		taskRun: taskRunPassTaskRefContentTempered,
+		wantErr: true,
+	}, {
+		name:    "Local taskRef Fail Verification with tempered content",
+		taskRun: taskRunPassTaskRefUnsigned,
+		wantErr: true,
+	},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			crd := tc.taskRun.DeepCopyObject().(resourcesemantics.GenericCRD)
+			gvk := schema.GroupVersionKind{
+				Group:   tc.taskRun.GetGroupVersionKind().Group,
+				Version: tc.taskRun.APIVersion,
+				Kind:    tc.taskRun.Kind,
+			}
+			err := rc.verifyResources(ctx, crd, gvk)
+			if (err != nil) != tc.wantErr {
+				t.Fatalf("verifyResources get err %v, wantErr %t", err, tc.wantErr)
+			}
+		})
+	}
+
+}
+
+func pushOCIImage(t *testing.T, u *url.URL, task *v1beta1.Task) (v1types.Hash, error) {
+	ref, err := remotetest.CreateImage(u.Host+"/task/"+task.Name, task)
+	if err != nil {
+		t.Errorf("uploading image failed unexpectedly with an error: %v", err)
+	}
+
+	imgRef, err := imgname.ParseReference(ref)
+	if err != nil {
+		t.Errorf("digest %s is not a valid reference: %v", ref, err)
+	}
+
+	img, err := remote.Image(imgRef, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+	if err != nil {
+		t.Errorf("could not fetch created image: %v", err)
+	}
+
+	dig, err := img.Digest()
+	if err != nil {
+		t.Errorf("failed to fetch img manifest: %v", err)
+	}
+	return dig, nil
+}
+
+func generateKeyFile(t *testing.T, tmpDir string, pf cosign.PassFunc) (privFile, pubFile string) {
+	t.Helper()
+
+	tmpPrivFile, err := os.Create(filepath.Join(tmpDir, "cosign.key"))
+	if err != nil {
+		t.Fatalf("failed to create temp key file: %v", err)
+	}
+	defer tmpPrivFile.Close()
+	tmpPubFile, err := os.Create(filepath.Join(tmpDir, "cosign.pub"))
+	if err != nil {
+		t.Fatalf("failed to create temp pub file: %v", err)
+	}
+	defer tmpPubFile.Close()
+
+	// Generate a valid keypair.
+	keys, err := cosign.GenerateKeyPair(pf)
+	if err != nil {
+		t.Fatalf("failed to generate keypair: %v", err)
+	}
+
+	if _, err := tmpPrivFile.Write(keys.PrivateBytes); err != nil {
+		t.Fatalf("failed to write key file: %v", err)
+	}
+	if _, err := tmpPubFile.Write(keys.PublicBytes); err != nil {
+		t.Fatalf("failed to write pub file: %v", err)
+	}
+	return tmpPrivFile.Name(), tmpPubFile.Name()
```

Comment:
Hmmm... it seems redundant to generate keys to write them to files to then read them the same files later. Would it be possible to remove the file requirement in the verification if we already have the keys loaded in memory?

Reply:
For current verifier I choose the similar from chains, read the cosign keys from mounted files. So for testing I 
still need to read pubkey from local file. I guess it should be ok to keep it?

Reply:
I'd have 2 versions of this - one that's completely in memory and one that writes out the file. For most of the verify funcs, we can actually just generate the keys in memory and pass back a SignerVerifier. It's only for testing the e2e flows when reading from a file that we need to write this out.

Reply:
I see, I will do it